tests/by-util/test_pmap.rs

```diff
@@ -211,27 +241,116 @@ fn assert_extended_format(pid: u32, s: &str) {
     let line_count = lines.len();
 
     let re = Regex::new(&format!("^{pid}:   .+[^ ]$")).unwrap();
-    assert!(re.is_match(lines[0]));
+    assert!(re.is_match(lines[0]), "failing line: '{}'", lines[0]);
 
     let expected_header = "Address           Kbytes     RSS   Dirty Mode  Mapping";
-    assert_eq!(expected_header, lines[1]);
+    assert_eq!(expected_header, lines[1], "failing line: '{}'", lines[1]);
 
     let re = Regex::new(
         r"^[0-9a-f]{16} +[1-9][0-9]* +\d+ +\d+ (-|r)(-|w)(-|x)(-|s)- (  \[ (anon|stack) \]|[a-zA-Z0-9._-]+)$",
     )
     .unwrap();
 
     for line in lines.iter().take(line_count - 2).skip(2) {
-        assert!(re.is_match(line), "failing line: {line}");
+        assert!(re.is_match(line), "failing line: '{line}'");
     }
 
     let expected_separator = "---------------- ------- ------- ------- ";
-    assert_eq!(expected_separator, lines[line_count - 2]);
+    assert_eq!(
+        expected_separator,
+        lines[line_count - 2],
+        "failing line: '{}'",
+        lines[line_count - 2]
+    );
 
     let re = Regex::new(r"^total kB +[1-9][0-9]* +\d+ +\d+$").unwrap();
     assert!(
         re.is_match(lines[line_count - 1]),
-        "failing line: {}",
+        "failing line: '{}'",
+        lines[line_count - 1]
+    );
+}
+
+// Ensure `s` has the following more extended format (-X):
+//
+// 1234:   /some/path
+//          Address Perm           Offset    Device      Inode Size  Rss  Pss Pss_Dirty Referenced Anonymous LazyFree ShmemPmdMapped FilePmdMapped Shared_Hugetlb Private_Hugetlb Swap SwapPss Locked THPeligible Mapping
+// 000073eb5f4c7000 r-xp 0000000000036000 008:00008    2274176 1284 1148 1148         0       1148         0        0              0             0              0               0    0       0      0           0 ld-linux-x86-64.so.2
```

Comment:
There are some differences with leading zeros compared to the output of the original `pmap`:

* for the `Address`, the original `pmap` doesn't show leading zeros, it uses spaces instead
* the length of the `Offset` is eight characters with the original `pmap`
* for the `Device`, the original `pmap` uses the `xx:xx` format

I don't know whether you want to fix it in this PR. As the values themselves are correct, it's also ok to add a todo.

Reply:
I implemented them to work for the default, extended (`-x`), and device (`-d`) formats. The more-extended and most-extended formats unfortunately differ in how some of the same data is shown.

Reply:
Sure, that's fine :)